python/llm/src/bigdl/llm/transformers/utils.py

```diff
@@ -133,3 +133,23 @@ def fix_key(key):
     if "gamma" in key:
         return key.replace("gamma", "weight")
     return key
+
+
+def check_autocast(x):
+    if x.device.type == "xpu":
+        return torch.xpu.is_autocast_xpu_enabled()
+    elif x.device.type == "cpu":
+        return torch.is_autocast_enabled()
+    else:
+        invalidInputError(False,
+                          f"Device {x.device} is not supported.")
+
+
+def get_autocast_dtype(x):
+    if x.device.type == "xpu":
+        return torch.xpu.get_autocast_xpu_dtype()
+    elif x.device.type == "cpu":
+        return torch.get_autocast_cpu_dtype()
+    else:
+        invalidInputError(False,
+                          f"Device {x.device} is not supported.")
```

Comment:
We can combine these functions into to one (e.g., `get_autocast_dtype()` returning `None` if not enabled) to reduce one conditional branch if needed

Reply:
Have revised to this way.